presto-main/src/main/java/io/prestosql/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -219,6 +210,15 @@ private boolean shouldMergeWithMemory(long memorySize)
         }
     }
 
+    private long getSizeInMemoryWhenUnspilling()
+    {
+        // TODO: we could skip memory reservation for hashAggregationBuilder.getGroupIdsSortingSize()
+        // if before building result from hashAggregationBuilder we would convert it to "read only" version.
+        // Read only version of GroupByHash from hashAggregationBuilder could be compacted by dropping
+        // most of it's field, freeing up some memory that could be used for sorting.
+        return hashAggregationBuilder.getSizeInMemory() + hashAggregationBuilder.getGroupIdsSortingSize();
```

Comment:
Why `hashAggregationBuilder.getSizeInMemory()` does not include `hashAggregationBuilder.getGroupIdsSortingSize()`?

Reply:
`hashAggregationBuilder.getGroupIdsSortingSize()` part is only materialized in memory when `buildHashSortedResult` is called. This happens only when unspilling. Therefore `getSizeInMemoryWhenUnspilling` is used to estimate aggregation size when unspilling only.